File: frontend/ios/Conq/Models/Content.swift
import Foundation

struct Content: Identifiable, Codable {
    let id: String
    let userId: String
    let title: String
    let body: String
    let mediaUrls: [String]?
    let tags: [String]?
    let status: ContentStatus
    let platforms: [String]?
    let createdAt: Date?
    let updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case userId = "user_id"
        case title
        case body
        case mediaUrls = "media_urls"
        case tags
        case status
        case platforms
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

enum ContentStatus: String, Codable {
    case draft
    case pending
    case approved
    case published
}

File: backend/services/content-service/models/content.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Content struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	UserID    string             `bson:"user_id" json:"user_id"`
	Title     string             `bson:"title" json:"title"`
	Body      string             `bson:"body" json:"body"`
	MediaURLs []string           `bson:"media_urls,omitempty" json:"media_urls,omitempty"`
	Tags      []string           `bson:"tags,omitempty" json:"tags,omitempty"`
	Status    string             `bson:"status" json:"status"` // draft, pending, approved, published
	Platforms []string           `bson:"platforms,omitempty" json:"platforms,omitempty"`
	CreatedAt time.Time          `bson:"created_at" json:"created_at"`
	UpdatedAt time.Time          `bson:"updated_at" json:"updated_at"`
}

type CreateContentRequest struct {
	UserID    string   `json:"user_id"`
	Title     string   `json:"title"`
	Body      string   `json:"body"`
	MediaURLs []string `json:"media_urls"`
	Tags      []string `json:"tags"`
	Platforms []string `json:"platforms"`
}
